pypeit/core/wavecal/autoid.py

```diff
@@ -687,6 +694,135 @@ def reidentify(spec, spec_arxiv_in, wave_soln_arxiv_in, line_list, nreid_min, de
     return detections, spec_cont_sub, patt_dict_slit
 
 
+def match_to_arxiv(lamps:list, spec:np.ndarray, wv_guess:np.ndarray,
+                   spec_arxiv:np.ndarray, wave_arxiv:np.ndarray, nreid_min:int,
+                   match_toler=2.0, nonlinear_counts=1e10, sigdetect=5.0, fwhm=4.0,
+                   debug_peaks:bool=False, use_unknowns:bool=False):
+    """ Algorithm to match an input arc spectrum to an archival arc spectrum
+    using a set wavelength guess for the input
+
+    Used only missing orders of echelle spectrographs (so far)
+
+    Args:
+        lamps (list): List of lamps used in the arc
+        spec (np.ndarray): Spectrum to match
+        wv_guess (np.ndarray): Wavelength guess for the input spectrum
+        spec_arxiv (np.ndarray): Spectrum to match to
+        wave_arxiv (np.ndarray): Wavelegnth solution for the archival spectrum
+        nreid_min (int): 
+            Minimum number of times that a given candidate reidentified line must be properly matched with a line in the arxiv
+            to be considered a good reidentification. If there is a lot of duplication in the arxiv of the spectra in question
+            (i.e. multislit) set this to a number like 2-4. For echelle this depends on the number of solutions in the arxiv.
+            For fixed format echelle (ESI, X-SHOOTER, NIRES) set this 1. For an echelle with a tiltable grating, it will depend
+            on the number of solutions in the arxiv.
+        match_toler (float, optional): Defaults to 2.0.
+            Matching tolerance in pixels for a line reidentification. A good line match must match within this tolerance to the
+            the shifted and stretched archive spectrum, and the archive wavelength solution at this match must be within
+            match_toler dispersion elements from the line in line list.
+        nonlinear_counts (float, optional): Defaults to 1e10.
+            For arc line detection: Arc lines above this saturation
+            threshold are not used in wavelength solution fits because
+            they cannot be accurately centroided
+        sigdetect (float, optional): Defaults to 5.0.
+            Threshold for detecting arcliens
+        fwhm (float, optional): Defaults to 4.0.
+            Full width at half maximum for the arc lines
+        debug_peaks (bool, optional): Defaults to False.
+        use_unknowns (bool, optional): Defaults to False.
+            If True, use the unknowns in the solution (not recommended)
+
+    Returns:
+        _type_: _description_
+    """
+
+    # TODO -- The next 10 lines of code is duplicated from echelle_wvcalib
```

Comment:
Should we make these 10 lines a method then?

Reply:
I went deep dive and refactored `load_line_lists` yet again.
  we'll see how the tests go..